packages/relay/tests/lib/services/metricService/metricService.spec.ts

```diff
@@ -0,0 +1,358 @@
+    });
+
+    it('Should execute captureTransactionMetrics() by retrieving transaction record from CONSENSUS NODE client', async () => {
+      process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE = 'true';
+      const mockedExchangeRateInCents = 12;
+      const expectedTxRecordFee = calculateTxRecordChargeAmount(mockedExchangeRateInCents);
+
+      const transactionRecordStub = sinon
+        .stub(TransactionRecordQuery.prototype, 'execute')
+        .resolves(mockedConsensusNodeTransactionRecord);
+
+      const originalBudget = hbarLimiter.getRemainingBudget();
+
+      await metricService.captureTransactionMetrics(mockedExecuteTransactionEventPayload);
+      expect(transactionRecordStub.called).to.be.true;
+
+      // validate hbarLimiter
+      // note: since the query is made to consensus node, the total charged amount = txFee + txRecordFee
+      const updatedBudget = hbarLimiter.getRemainingBudget();
+      expect(originalBudget - updatedBudget).to.eq(mockedTxFee + expectedTxRecordFee);
+
+      // validate cost metrics
+      const costMetricObject = (await metricService.getCostMetric().get()).values.find(
+        (metric) => metric.metricName === metricHistogramCostSumTitle,
+      )!;
+      expect(costMetricObject.metricName).to.eq(metricHistogramCostSumTitle);
+      expect(costMetricObject.labels.caller).to.eq(mockedCallerName);
+      expect(costMetricObject.labels.interactingEntity).to.eq(mockedInteractingEntity);
+      expect(costMetricObject.value).to.eq(mockedTxFee + expectedTxRecordFee);
+
+      // validate gas metric
+      const gasMetricObject = (await metricService.getGasFeeMetric().get()).values.find(
+        (metric) => metric.metricName === metricHistogramGasFeeSumTitle,
+      )!;
+
+      expect(gasMetricObject.metricName).to.eq(metricHistogramGasFeeSumTitle);
+      expect(gasMetricObject.labels.caller).to.eq(mockedCallerName);
+      expect(gasMetricObject.labels.interactingEntity).to.eq(mockedInteractingEntity);
+      expect(gasMetricObject.value).to.eq(
+        mockedConsensusNodeTransactionRecord.contractFunctionResult?.gasUsed.toNumber(),
+      );
+    });
+
+    it('Should listen to EXECUTE_TRANSACTION event to kick off captureTransactionMetrics()', async () => {
+      process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE = 'true';
+      const mockedExchangeRateInCents = 12;
+      const expectedTxRecordFee = calculateTxRecordChargeAmount(mockedExchangeRateInCents);
+
+      const transactionRecordStub = sinon
+        .stub(TransactionRecordQuery.prototype, 'execute')
+        .resolves(mockedConsensusNodeTransactionRecord);
+
+      const originalBudget = hbarLimiter.getRemainingBudget();
+
+      // emitting an EXECUTE_TRANSACTION event to kick off capturing metrics process asynchronously
+      eventEmitter.emit(constants.EVENTS.EXECUTE_TRANSACTION, mockedExecuteTransactionEventPayload);
+
+      // small wait for hbar rate limiter to settle
+      await new Promise((r) => setTimeout(r, 100));
```

Comment:
```suggestion
      await new Promise((r) => eventEmitter.once(constants.EVENTS.EXECUTE_TRANSACTION, r));
```

Reply:
Oh nvm, we also need to wait for the addExpenseAndCaptureMetrics method to finish executing, disregard this comment

Reply:
the suggestion failed with 
```bash
Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```